### PR TITLE
Deal with NPEs

### PR DESCRIPTION
**Pull request by ScoreUnder**
I added nullity annotations, checked where they conflict, and then went and fixed things.

Some larger changes:

* The `get(criteria)` methods were unused and needed tweaking to work after these changes, so I removed them (I can probably go back on this if you had big plans for that).
* Restructured commands so that the fields don't need to be assigned to. (Easier to ensure non-null)
* Made WatchedUser/WatchedRole/ExcludedChannel operate on IDs only, not requiring JDA's agreement on whether or not those entities are retrievable right now. (Prevents nulls and/or exceptions from creeping in if we can't get a corresponding object from JDA's cache)

There were a couple of minor bugs like missing `queue()`, creation of a variable and accidentally not using it, etc. which I also zapped along the way.

---

Also, since it's not entirely obvious at first glance if you haven't read it before, `/*1*/@NonNull Thing /*2*/@NonNull []` means a non-null array (2) of non-null Things (1).

### CODE REVIEW

File: src/main/java/pw/byakuren/discord/util/BotEmbed.java [hunk 1]
```diff
@@ -15,51 +17,51 @@
...
                         .collect(Collectors.joining("\n")));
     }
 
-    public static EmbedBuilder footerAuthor(String msg, User u) {
+    public static @NotNull EmbedBuilder footerAuthor(@Nullable String msg, @NotNull User u) {
         return neutral(msg).setFooter(u.getName()+"#"+u.getDiscriminator(), u.getEffectiveAvatarUrl());
     }
 
-    public static EmbedBuilder footerAuthor(String msg, User u, Color c) {
+    public static @NotNull EmbedBuilder footerAuthor(@Nullable String msg, @NotNull User u, @Nullable Color c) {
```
Comment (Brod8362) [inline src/main/java/pw/byakuren/discord/util/BotEmbed.java:48]: Are these params really nullable? (Need to check further down the call stack)

File: src/main/java/pw/byakuren/discord/modules/ModuleInfo.java [hunk 2]
```diff
@@ -1,13 +1,15 @@
...
-    public final String author;
-    public  final String version;
-    public final ModuleType type;
+    public final @NotNull String name;
+    public final @NotNull String author;
+    public final @NotNull String version;
+    public final @NotNull ModuleType type;
 
-    public ModuleInfo(String name, String author, String version, ModuleType type) {
+    public ModuleInfo(@NotNull String name, @NotNull String author, @NotNull String version, @NotNull ModuleType type) {
```
Comment (Brod8362) [inline src/main/java/pw/byakuren/discord/modules/ModuleInfo.java:12]: I think there may have been some intention for some of these parameters to be able to be null.

**Review by Brod8362 — COMMENTED** [review]
There are a few places that need to be checked for original intent, e.g it may be possible the values may have allowed to be null. (Without proper documentation it's hard to say, will need to check all uses)

File: src/main/java/pw/byakuren/discord/util/BotEmbed.java (hunk 1 above)
Comment (ScoreUnder) [reply src/main/java/pw/byakuren/discord/util/BotEmbed.java:48]: I checked the relevant JDA docs, so they're nullable if it's technically supported by JDA

File: src/main/java/pw/byakuren/discord/modules/ModuleInfo.java (hunk 2 above)
Comment (ScoreUnder) [reply src/main/java/pw/byakuren/discord/modules/ModuleInfo.java:12]: is this a YAGNI

**Comment by ScoreUnder** [comment]
> There are a few places that need to be checked for original intent, e.g it may be possible the values may have allowed to be null. (Without proper documentation it's hard to say, will need to check all uses)

IntelliJ can use the annotations to determine if there is a null being passed into a non-null variable, so you can just run a static analysis

**Comment by Brod8362** [comment]
Closed due to EOL.

**Pull request closed without merging**